pkg/typesystem/typesystem.go

```diff
@@ -621,8 +632,12 @@ func (t *TypeSystem) allRelations() map[string]*openfgapb.Relation {
 //     a. For a type (e.g. user) this means checking that this type is in the *TypeSystem
 //     b. For a type#relation this means checking that this type with this relation is in the *TypeSystem
 //  4. Check that a relation is assignable if and only if it has a non-zero list of types
-func NewAndValidate(model *openfgapb.AuthorizationModel) (*TypeSystem, error) {
-	t := New(model)
+func NewAndValidate(model *openfgapb.AuthorizationModel, allowSchema10 bool) (*TypeSystem, error) {
```

Comment:
This and the `NewAndNonObsolete` code above feel out of place in this package. Let's not change this signature.


The logic around allowing or disallowing 1.0 models shouldn't not belong to the `typesystem` package. The typesystem package is meant to work with the typesystem of a model, but not to enforce the model validation/typesystem rules. That's the caller's responsibility.

The server implementation of the gRPC API should be the responsible party for enforcing the business logic for the API (whether we put that in `commands` or in the server RPC implementations themselves is up for debate).

I think the approach we had before at the server layer was reasonable.

Reply:
Done